tests/regression/test_matrix_free.py

```diff
@@ -308,3 +308,34 @@ def test_duplicate(a, bcs):
         B_petsc.mult(x, y)
     # Check if original rhs is equal to BA^-1 (rhs)
     assert np.allclose(rhs.vector().array(), solution2.vector().array())
+
+
+def test_matrix_free_fieldsplit_with_real():
```

Comment:
Is this really matrix free? I can't see anything in the solver options that would make it so.

Reply:
Turned out that, when we have `Real`, `solving_utils.set_defaults()` puts `matfree` here: https://github.com/firedrakeproject/firedrake/blob/06125881584cc898fed1f9b9c37a25bbe3bb55c9/firedrake/solving_utils.py#L95

Reply:
Perhaps do an assertion for `stokes_solver.parameters["mat_type"]` to make this explicit?

Reply:
Or can I just add comments, as `set_defaults()` also sets many related `fieldsplit` options?

Reply:
Actually, I agree. It's probably better to spell out options we need here; updated.